### DISCUSSION TITLE
How to get user entity?

### DISCUSSION THREAD

**Discussion by npabisz** [saul/demofile#205] [Questions] [answered]
So how can I get `DT_BasePlayer` based on `userinfo` table data? Is there corresponding field I should look for? Since I cant find steam id in `DT_BasePlayer`.

**Comment by saul** [comment, marked as answer]
What are you trying to do? Userinfo table has steam ID. 

If you have a userinfo entry and want to get the player entity (DT_BasePlayer), do demo.entities.entities[userinfoIndex + 1]. It is always the case that the entity for a particular userinfo is the same index + 1. Userinfo 0 is the first player, entity index 1 is the first player. 

The corollary is also true - if you want to get the userinfo for a player, look at the entity.index - 1 entry in the userinfo string table.

**Comment by npabisz** [comment]
Thats great, just wanted to combine these two datas so I could identify player entities based on steam id since I'm making demo parser to extract some useful data :)
